app/scripts/modules/cloudfoundry/src/serverGroup/configure/serverGroupCommandBuilder.service.cf.ts

```diff
@@ -93,9 +93,8 @@ export class CloudFoundryServerGroupCommandBuilder {
         routes: serverGroup.loadBalancers,
         environment: CloudFoundryServerGroupCommandBuilder.envVarsFromObject(serverGroup.env),
         services: (serverGroup.serviceInstances || []).map(serviceInstance => serviceInstance.name),
-        reference: '',
```

Comment:
After adding the discriminator, these fields `reference`, `account`, and `pattern` were flagged as a type error.  It also told me that `healthCheckType` and `healthCheckHttpEndpoint` were required when `type === 'direct'`.  Can you validate that this is correct?